test/writev.js

```diff
@@ -0,0 +1,108 @@
+      // Send one less span than bufferWindowSize
+      for (let n = 1; n <= 50; n++) {
+        client[sendFn]({req: n})
+        t.ok(client._writableState.corked, 'should be corked')
+      }
+
+      // This span should trigger the uncork
+      client[sendFn]({req: 51})
+
+      // Wait a little to allow the above write to finish before destroying
+      process.nextTick(function () {
+        t.notOk(client._writableState.corked, 'should be uncorked')
+
+        client.destroy()
+        server.close()
+        t.end()
+      })
+    })
+  })
+
+  test(`bufferWindowSize - custom value (${dataType})`, function (t) {
+    const server = APMServer().client({bufferWindowSize: 5}, function (client) {
+      // Send one less span than bufferWindowSize
+      for (let n = 1; n <= 5; n++) {
+        client[sendFn]({req: n})
+        t.ok(client._writableState.corked, 'should be corked')
+      }
+
+      // This span should trigger the uncork
+      client[sendFn]({req: 6})
+
+      // Wait a little to allow the above write to finish before destroying
+      process.nextTick(function () {
+        t.notOk(client._writableState.corked, 'should be uncorked')
+
+        client.destroy()
+        server.close()
+        t.end()
+      })
+    })
+  })
+
+  test(`bufferWindowTime - default value (${dataType})`, function (t) {
+    const server = APMServer().client(function (client) {
+      client[sendFn]({req: 1})
+      t.ok(client._writableState.corked, 'should be corked')
+
+      // Wait twice as long as bufferWindowTime
+      setTimeout(function () {
+        t.notOk(client._writableState.corked, 'should be uncorked')
+        client.destroy()
+        server.close()
+        t.end()
+      }, 40)
+    })
+  })
+
+  test(`bufferWindowTime - custom value (${dataType})`, function (t) {
+    const server = APMServer().client({bufferWindowTime: 150}, function (client) {
+      client[sendFn]({req: 1})
+      t.ok(client._writableState.corked, 'should be corked')
+
+      // Wait twice as long as the default bufferWindowTime
+      setTimeout(function () {
+        t.ok(client._writableState.corked, 'should be corked')
+      }, 40)
+
+      // Wait twice as long as the custom bufferWindowTime
+      setTimeout(function () {
+        t.notOk(client._writableState.corked, 'should be uncorked')
+        client.destroy()
+        server.close()
+        t.end()
+      }, 300)
+    })
+  })
+
+  test(`write on destroyed (${dataType})`, function (t) {
+    const server = APMServer().client({bufferWindowSize: 1}, function (client) {
+      client.on('error', function (err) {
+        t.ok(err instanceof Error)
+        t.equal(err.message, 'write called on destroyed Elastic APM client')
+        server.close()
+        t.end()
+      })
+
+      client[sendFn]({req: 1})
+      client[sendFn]({req: 2})
+
+      // Destroy the client before the _writev function have a chance to be called
+      client.destroy()
```

Comment:
@Qard I'm actually not sure this is a feature, but instead a bug? What would you expect to happen in this case? Should just abort writing without emitting the error?

Reply:
I'm not sure what you are trying to do here...

Reply:
It's difficult to reason about the exact timing...

Is it possible that it might get aborted part way through a `_writev`? As in, some of the data has been written, but not all? How would we want to react to an abort? Retrying? Dropping the data?

I don't really have an informed opinion on what to do about this right now.

Reply:
I thought about it some more and decided to swallow this error